src/mesh/gmsh_io.C

```diff
@@ -290,20 +384,62 @@ void GmshIO::read_mesh(std::istream & in)
 
               // add the nodal coordinates to the mesh
               for (unsigned int i=0; i<num_nodes; ++i)
+              {
+                in >> id >> x >> y >> z;
+                mesh.add_point (Point(x, y, z), i);
+                nodetrans[id] = i;
+              }
+            }
+            else
+            {
+              // Read numEntityBlocks line
+              size_t num_entities = 0, num_nodes = 0, min_node_tag, max_node_tag;
+              in >> num_entities >> num_nodes >> min_node_tag >> max_node_tag;
+
+              mesh.reserve_nodes(num_nodes);
+
+              size_t node_counter = 0;
+
+              // Now loop over entities
+              for (size_t i = 0; i < num_entities; ++i)
+              {
+                int entity_dim, entity_tag, parametric;
+                size_t num_nodes_in_block = 0;
+                in >> entity_dim >> entity_tag >> parametric >> num_nodes_in_block;
+                libmesh_assert_msg(!parametric, "We don't currently support reading parametric gmsh entities");
```

Comment:
Should this be a `libmesh_error_msg()` as well? If this is a parametric entity, presumably something else is going to fail later on and it's better to get a simple error now rather than a cryptic error later...